commit 649298dcf84b2f20cfe458d887c1591db47372a6
Author: yuan <zodf0055980@gmail.com>
Date:   Wed Nov 25 20:41:39 2020 +0800

    Encoder: grow again buffer size in opj_tcd_code_block_enc_allocate_data() (fixes #1283)

diff --git a/src/lib/openjp2/tcd.c b/src/lib/openjp2/tcd.c
index cb305a63..edc2f416 100644
--- a/src/lib/openjp2/tcd.c
+++ b/src/lib/openjp2/tcd.c
@@ -1246,10 +1246,12 @@ static OPJ_BOOL opj_tcd_code_block_enc_allocate_data(opj_tcd_cblk_enc_t *
     /* and +7 for https://github.com/uclouvain/openjpeg/issues/1283 (-M 3) */
     /* and +26 for https://github.com/uclouvain/openjpeg/issues/1283 (-M 7) */
     /* and +28 for https://github.com/uclouvain/openjpeg/issues/1283 (-M 44) */
+    /* and +33 for https://github.com/uclouvain/openjpeg/issues/1283 (-M 4) */
+    /* and +63 for https://github.com/uclouvain/openjpeg/issues/1283 (-M 4 -IMF 2K) */
     /* TODO: is there a theoretical upper-bound for the compressed code */
     /* block size ? */
-    l_data_size = 28 + (OPJ_UINT32)((p_code_block->x1 - p_code_block->x0) *
-                                   (p_code_block->y1 - p_code_block->y0) * (OPJ_INT32)sizeof(OPJ_UINT32));
+    l_data_size = 63 + (OPJ_UINT32)((p_code_block->x1 - p_code_block->x0) *
+                                    (p_code_block->y1 - p_code_block->y0) * (OPJ_INT32)sizeof(OPJ_UINT32));
 
     if (l_data_size > p_code_block->data_size) {
         if (p_code_block->data) {
